crates/adapters/src/transport/kafka/input.rs

```diff
@@ -1,4 +1,5 @@
 use super::refine_kafka_error;
+use crate::transport::secret_resolver::SimpleOrSecretString;
```

Comment:
Looks good overall!

* Could we just have one enum type called `MaybeSecret`, which has three entries: `String`, `SecretRef(String)` and `Secret(String)`? Secret resolution just produces a new instance that converts SecretRefs to Secrets. The "Simple" qualifier doesn't add much.

* If we can't avoid the two separate types, could we just make it less wordy and go with `MaybeSecret` and `MaybeSecretRef`?

Reply:
- The separation into the two types is more for later purposes when one of them (`MaybeSecretRef`) would become part of the API, and the other (`MaybeSecret`) after it has been resolved.
-Yes, naming them `MaybeSecret::{String(String), Secret(String)}` and `MaybeSecretRef::{String(String), SecretRef(String)}` would be the way to go then?

Reply:
> Yes, naming them MaybeSecret::{String(String), Secret(String)} and MaybeSecretRef::{String(String), SecretRef(String)} would be the way to go then?

Yes, sounds good.

Reply:
Updated naming.